missing-dependency/src/test/java/MissingDependencyTest.java

```diff
@@ -0,0 +1,11 @@
+import org.eclipse.collections.api.factory.list.Lists;
+import org.junit.Test;
+
+public class MissingDependencyTest
+{
+    @Test
+    public void missingDependency()
+    {
+        Lists.mutable.empty();
```

Comment:
With the dynamic proxy, the error message shows the line which calls empty(), but no longer includes the method empty().

IntelliJ collapses the error message to:
```
java.lang.IllegalStateException: Could not find any implementations of MutableListFactory. Check that eclipse-collections.jar is on the classpath and that its META-INF/services directory is intact.

	at org.eclipse.collections.api.factory.ThrowingInvocationHandler.invoke(ThrowingInvocationHandler.java:18)
	at MissingDependencyTest.missingDependency(MissingDependencyTest.java:9)
```

The full stack trace is:
```
java.lang.IllegalStateException: Could not find any implementations of MutableListFactory. Check that eclipse-collections.jar is on the classpath and that its META-INF/services directory is intact.

	at org.eclipse.collections.api.factory.ThrowingInvocationHandler.invoke(ThrowingInvocationHandler.java:18)
	at com.sun.proxy.$Proxy5.empty(Unknown Source)
	at MissingDependencyTest.missingDependency(MissingDependencyTest.java:9)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
```

Reply:
I think this is pretty nice.